app/models/todo_item.rb

```diff
@@ -1,7 +1,35 @@
 class TodoItem < ApplicationRecord
   belongs_to :todo_list
 
+  after_create :assign_status
+
+  enum status: {red: 0, yellow: 1, blue: 2, lightgreen: 3}
+
+  default_scope { order(due_date: :asc) }
+
   def completed?
     !completed_at.blank?
   end
+
+  def item_complete_updation
+    self.update!(completed_at: Time.now, status: :lightgreen, due_date: close_to_infinite)
```

Comment:
Why have you set due_date to be close_to_infinite upon completion? Should due date be even automatically triggered?
Didn't understand the use case here. Please add comment

Reply:
Actually What I am trying I am trying to keep the tasks as per the order of due date. So when task got completed. In that case I am just keeping the task in the end. So that's why keeping it as close_to_infinite